src/lightning/fabric/utilities/spike.py

```diff
@@ -0,0 +1,165 @@
+import json
+import operator
+import os
+import warnings
+from typing import Any, Dict, List, Literal, Optional, type_checking, Union
+
+import torch
+from lightning_utilities.core.imports import compare_version
+
+_TORCHMETRICS_GREATER_EQUAL_1_0_0 = compare_version("torchmetrics", operator.ge, "1.0.0")
+
+if type_checking:
+    from lightning.fabric.fabric import Fabric
+
+
+class SpikeDetection:
+    """Spike Detection Callback.
+
+    Terminates training with a ``TrainingSpikeException`` when a loss-spike was detected and
+    saves the batches to skip when resuming to a file.
+
+    Currently we skip the current and the previous batch since it is unclear, whether the previous batch
+    altered the weights in a way that it causes the spike or just the current batch is corrupted somehow.
+
+    Args:
+        mode: Whether to minimize or maximize the tracked metric
+        window: A running mean of metrics with ``window`` size. Serves as reference value for spike3s
+        warmup: After how many batches should spike-tracking starts
+        atol: An absolute tolerance.  Every diff between the running mean and the current value,
+            that's not an improvement and above ``atol`` will be considered a spike
+        rtol: A relative tolerance. Every diff between the running mean and the current value,
+            that's higher than ``rtol * running_mean`` is considered a spike
+        exclude_batches_path: Where to save the file that contains the batches to exclude. Will default to current directory.
+        finite_only: Will consider non-finite values like NaN, inf and -inf a spike as well
```

Comment:
That's awesome, I like this feature!
Note that the description is the opposite of what the argument does by default. So maybe change to:
```suggestion
        finite_only: If set to ``False``, consider non-finite values like NaN, inf and -inf a spike as well.
```